### PR TITLE
[Refactor][Admin]Refactor Add Global Sse

### PR DESCRIPTION
**Pull request by gaoyan1998**
# How to use

The java
```java
//SseTopic.java
public enum SseTopic {

    METRICS("/TOPIC/METRICS"),
    PROCESS_CONSOLE("/TOPIC/PROCESS_CONSOLE"),
    PRINT_TABLE("/TOPIC/PRINT_TABLE"),
    UNKNOWN("UNKNOWN");
  ...
}

//other java that need sse

SseSessionContextHolder.sendTopic( SseTopic.METRICS.getValue(), "{jsonData}");

```

The react
```typescript

  const {subscribeTopic} = useModel('Sse',(model:any)=>({subscribeTopic:model.subscribeTopic}))

useEffect(() => {
  const topic = `${SSE_TOPIC.PRINT_TABLE}/${title.fullTableName}`
   //Must reurn
   return subscribeTopic([topic],(data:SseData)=>{
        setConsoleInfo((preConsoleInfo) => preConsoleInfo + '\n' + data.data);
   });
  }, []);


```

### CODE REVIEW

File: dinky-web/src/models/Sse.tsx [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+        subscriberRef.current
+          .filter(sub => sub.topic.includes(data.topic))
+          .forEach(sub => sub.call(data));
+      }
+    }
+  }, [eventSource]);
+
+  const subscribeTopic = (topic: string[], onMessage: (data: SseData) => void) => {
+    const sub:SubscriberData = {topic: topic, call: onMessage}
+    subscriberRef.current = [sub, ...subscriberRef.current];
```
Comment (leechor) [inline dinky-web/src/models/Sse.tsx]: it should be [subscriberRef.current, sub], sub could over old value if exists

**Review by leechor — CHANGES_REQUESTED** [review]

File: dinky-web/src/models/Sse.tsx (hunk 1 above)
Comment (gaoyan1998) [reply dinky-web/src/models/Sse.tsx]: ok, Done.

**Review by leechor — APPROVED** [review]
LGTM

**Review by zackyoungh — APPROVED** [review]
LGTM

**Pull request merged**